api/pipeline.go

```diff
@@ -0,0 +1,647 @@
+// Copyright (c) 2020 Target Brands, Inc. All rights reserved.
+//
+// Use of this source code is governed by the LICENSE file in this repository.
+
+package api
+
+import (
+	"fmt"
+	"net/http"
+	"strings"
+
+	"github.com/go-vela/compiler/compiler"
+
+	"github.com/go-vela/server/database"
+	"github.com/go-vela/server/router/middleware/repo"
+	"github.com/go-vela/server/source"
+	"github.com/go-vela/server/util"
+
+	"github.com/go-vela/types"
+	"github.com/go-vela/types/yaml"
+
+	"github.com/gin-gonic/gin"
+)
+
+const (
+	outputJSON = "json"
+	outputYAML = "yaml"
+)
+
+// swagger:operation GET /api/v1/pipelines/{org}/{repo} pipelines GetPipeline
+//
+// Get a pipeline configuration from the source provider
+//
+// ---
+// x-success_http_code: '200'
+// produces:
+// - application/json
+// parameters:
+// - in: path
+//   name: repo
+//   description: Name of the repo
+//   required: true
+//   type: string
+// - in: path
+//   name: org
+//   description: Name of the org
+//   required: true
+//   type: string
```

Comment:
I noticed we don't mention the query parameters here.. should we?

Since we support two query parameters (`output`, `ref`), I'm thinking it might look like:

```suggestion
// parameters:
// - in: path
//   name: repo
//   description: Name of the repo
//   required: true
//   type: string
// - in: path
//   name: org
//   description: Name of the org
//   required: true
//   type: string
// - in: query
//   name: output
//   description: format of the output returned ( json | yaml )
//   required: false
//   type: string
// - in: query
//   name: ref
//   description: source reference to fetch pipeline from
//   required: false
//   type: string
```